### PR TITLE
gui: Prefer standard library algorithms

### PR DESCRIPTION
**Pull request by tcoyvwac**
Cleans up and abstracts some ranged-for loops for their standard library equivalent.

Happy to condense / squash into one commit. :smile_cat: 
Currently PR is in this commit form to show changeset history (before squash / merge).

### CODE REVIEW

File: vita3k/gui/src/content_manager.cpp [hunk 1]
```diff
@@ -29,6 +31,19 @@
...
+namespace {
+template <typename T>
+auto get_recursive_directory_size(const T &path) {
+    const auto &path_list = fs::recursive_directory_iterator(path);
+    const auto pred = [](const auto acc, const auto &app) {
+        if (fs::is_regular_file(app.path()))
+            return acc + fs::file_size(app.path());
+        return acc;
+    };
+    return boost::accumulate(path_list, 0ul, pred);
```
Comment (hobyst) [inline vita3k/gui/src/content_manager.cpp:43]: Increasing dependency on Boost when `std::accumulate` already exists is redundant. Are there any advantages over using the `std` implementation?

File: vita3k/gui/src/content_manager.cpp [hunk 2]
```diff
@@ -590,9 +590,7 @@ void draw_content_manager(GuiState &gui, HostState &host) {
                     content.second = false;
             }
         }
-        const auto is_enable = std::find_if(contents_selected.begin(), contents_selected.end(), [&](const auto &cs) {
-            return cs.second;
-        }) != contents_selected.end();
+        const auto is_enable = boost::algorithm::any_of(contents_selected, [&](const auto &cs) { return cs.second; });
```
Comment (hobyst) [inline vita3k/gui/src/content_manager.cpp:593]: Increasing dependency on Boost when `std::algorithm::any_of` already exists is redundant. Are there any advantages over using the `std` implementation?

**Comment by tcoyvwac** [comment]
I am happy to remove `boost` algorithms marked and use the `std` version.
Currently, the only reason for using the `boost` version is for code clarity / minimalism.

For future PRs, does this project accept using `boost::range`? This is quite a useful library which can do things that the `std` library cannot do (easily or at all).

**Comment by hobyst** [comment]
> I am happy to remove `boost` algorithms marked and use the `std` version. Currently, the only reason for using the `boost` version is for code clarity / minimalism.
> 
> For future PRs, does this project accept using `boost::range`? This is quite a useful library which can do things that the `std` library cannot do (easily or at all).

As far as I know, there's no explicit policy on what implementations to use. But regarding to Boost, keep in mind that while CI builds use standard and already-compiled Boost distributions, developer builds and environments use [`Vita3K/ext-boost`](https://github.com/Vita3K/ext-boost). Check if the Boost library you want to use it's there and if it doesn't then PR to `ext-boost` adding the library before using it in Vita3K.

File: vita3k/gui/src/content_manager.cpp (hunk 1 above)
Comment (tcoyvwac) [reply vita3k/gui/src/content_manager.cpp:43]: For this function & case, there is one advantage / difference:
* `fs::recursive_directory_iterator` is accepted as a "first-class citizen" datatype and the special `fs` iterators are accepted "as is", just like `std` iterators.
* In the `std` version, you must use special `fs::{begin, end}()` free functions for `fs::recursive_directory_iterator` to work.

In this case, the `boost` version is very streamlined & minimalistic to use, making it less error-prone and understandable.

File: vita3k/gui/src/content_manager.cpp (hunk 2 above)
Comment (tcoyvwac) [reply vita3k/gui/src/content_manager.cpp:593]: There is no difference compared to the `std` version except its function signature / parameters.

**Comment by tcoyvwac** [comment]
Humm, any reason for the failing build on Windows? It suggests something wrong with the `CMake` part of the CI, but I cannot see further detail & information.

**Comment by Zangetsu38** [comment]
> Humm, any reason for the failing build on Windows? It suggests something wrong with the `CMake` part of the CI, but I cannot see further detail & information.

You need just look end log in compile, it said error caused it

**Comment by tcoyvwac** [comment]
Ahh, was logged in, looking at the logs, but it was blank when checking for more details! ...But it seems I was having problems with GitHub's JS! :sweat_smile: Oh GitHub. Thanks for the assist @Zangetsu38 .

**Comment by Zangetsu38** [comment]
for error
image

**Review by pent0 — APPROVED** [review]
Good place to use these functions, thanks! If you are good to merge please tell.

**Comment by tcoyvwac** [comment]
Thanks for the encouragement and review! :smile_cat: I am happy for the PR to be (squashed? and) merged.

**Comment by Zangetsu38** [comment]
> Thanks for the encouragement and review! 😸 I am happy for the PR to be (squashed? and) merged.

Sorry is done totaly forget ^^

**Comment by Zangetsu38** [comment]
@tcoyvwac master is now broken after this pr
master branch no have include <boost/range/numeric.hpp>

**Pull request merged**